nuxeo-page-provider-display-behavior.html

```diff
@@ -426,23 +351,119 @@
         this.$.list.scrollToItem(item);
       },
 
-      scrollToIndex: function(idx) {
-        this.$.list.scrollToIndex(idx);
+      scrollToIndex: function(index) {
+        this.$.list.scrollToIndex(Math.min(Math.max(index, 0), this.items.length - 1));
+      },
+
+      focusOnIndexIfNotVisible: function(index) {
+        if (!this.$.list._isIndexVisible(index)) {
+          this.$.list.scrollToIndex(index);
+        }
       },
 
       _observeEmptyLabel: function() {
         this._computedEmptyLabel = this.emptyLabel;
       },
 
-      focusOnIndexIfNotVisible: function(idx) {
-        if (!this.$.list._isIndexVisible(idx)) {
-          this.$.list.scrollToIndex(idx);
+      _quickFilterChanged: function() {
+        this.fetch(0, this.nxProvider.pageSize, true);
+      },
+
+      _updateFlags: function() {
+        if (!this.loading) {
+          var selectedItems = this.$.list.selectedItems;
+          this._isSelectAllIndeterminate = selectedItems
+            && selectedItems.length > 0
+            && selectedItems.length < this.size;
+          this._isSelectAllChecked = selectedItems && selectedItems.length > 0 && selectedItems.length === this.size;
+          this._isEmpty = this.size === 0;
+          this._computedEmptyLabel = (this.filters && this.filters.length > 0) ?
+            this.emptyLabelWhenFiltered : this.emptyLabel;
+        }
+      },
+
+      reset: function(size) {
+        this.set('items', []);
+        if (size) {
+          for (var i = 0; i < size; i++) {
+            this.push('items', {});
```

Comment:
The profiler reveals this to be where most of the time is spent, but not doing so will prevent the iron list from being correctly refreshed.

Reply:
Also a significant time is spent in the ```_increasePool``` function in the ```iron-list``` element.